adding artic version 1.6.2

erinyoung opened the pull request:
There's a new version of [artic](https://github.com/artic-network/fieldbioinformatics/releases/tag/v1.5.8)! 

Actually, there have been a few versions.

The usage has changed, so I've updated the base image version and the tests. I've also added in a new ENV variable.

Notable changes:
medaka and nanopolish support have been dropped. It's all Clair3 now.
medaka models are downloaded into the artic path
non-artic bedfiles and references are easier to use
artic-tools has been dropped

In short, any workflow using the 'latest' tag is going to fail. 


Pull Request (PR) checklist:
- [X] Include a description of what is in this pull request in this message.
- [X] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [X] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [X] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [X] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [X] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [X] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [X] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [X] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

erinyoung commented:
Looks like there are tensorflow errors. I will address these and then take this PR out of draft.

kapsakcj commented:
I'm not seeing any tensorflow issues but the output is quite verbose so I could have missed it. I got the image to build locally.

I made some changes because there was a lot of stuff installed via `apt` that was unnecessary. micromamba handles all the dependencies required so it should reduce the size of the image.

I also added a list of important dependencies to the readme for artic

Also...v1.6.1 of artic was released and it would be good to add. Perhaps we can do that in a separate PR after this one is polished up, merged and deployed? https://github.com/artic-network/fieldbioinformatics/releases/tag/v1.6.1

I won't approve and merge until you've had some time to review and let me know about the tensorflow issues you mentioned

erinyoung commented:
@kapsakcj , my apologies for taking so long to get back to this!!! Thank you for your changes!!!

I've decided to forgo version 1.5.8 in favor of 1.6.1 (the current version).

I've updated all the files to reflect that version.

I also added some hyperlinks so all the artic* images have the correct hyperlinks.

erinyoung commented:
IT WORKS!!!

```
#15 [test 6/8] RUN artic minion --normalise 200 --threads 4  --model ont --read-file SRR22452250_1_filtered.fastq.gz --scheme-name sars-cov-2  --scheme-version  V5.3.2 test2 &&     ls test2*
#15 0.885 Running: samtools faidx /test/primer-schemes/artic-sars-cov-2/400/v5.3.2/reference.fasta
#15 0.889 Running: minimap2 -a -x map-ont -t 4 /test/primer-schemes/artic-sars-cov-2/400/v5.3.2/reference.fasta SRR22452250_1_filtered.fastq.gz | samtools view -bS -F 4 - | samtools sort -o test2.sorted.bam -
#15 49.44 full_alignment.vcf.gz.tbi
#15 49.44 log
#15 49.44 merge_output.vcf.gz
#15 49.44 merge_output.vcf.gz.tbi
#15 49.44 pileup.vcf.gz
#15 49.44 pileup.vcf.gz.tbi
#15 49.44 run_clair3.log
#15 49.44 tmp
#15 49.44 
#15 49.44 test2_rg_2:
#15 49.44 full_alignment.vcf.gz
#15 49.44 full_alignment.vcf.gz.tbi
#15 49.44 log
#15 49.44 merge_output.vcf.gz
#15 49.44 merge_output.vcf.gz.tbi
#15 49.44 pileup.vcf.gz
#15 49.44 pileup.vcf.gz.tbi
#15 49.44 run_clair3.log
#15 49.44 tmp
```

erinyoung commented:
@kapsakcj , I've updated this for version 1.6.2. This should be ready for review now.

kapsakcj commented on build-files/artic/1.6.2/README.md:
```diff
@@ -0,0 +1,319 @@
+
+WARNING : This container does not contain the primer schemes found at [https://github.com/artic-network/primer-schemes](https://github.com/artic-network/primer-schemes). Those will have to be downloaded and mounted separately. There are now commands to download the use the primers at runtime.
+
+This image DOES contain the medaka models downloaded with `artic_get_models`
```
is it accurate to call these model files "medaka" models?

I thought medaka was deprecated in favor of Clair3, right?

If so, wouldn't we refer to them as "Clair3" models?

more info on Sam's blog post here: https://virological.org/t/fieldbioinformatics-updates/988

There's mentions throughout the Dockerfile and this readme that I think should be updated to say "Clair3 models" instead

erinyoung replied:
The models are Clair3 models (they have URLs like https://cdn.oxfordnanoportal.com/software/analysis/models/clair3/r1041_e82_260bps_fast_g632.tar.gz ) , so I've adjusted the documentation. Sorry!

kapsakcj reviewed (changes requested):
tests pass and most everything looks great, but I think we should change the language around "Clair3 models" in the Dockerfile and artic README since (I think) it's inaccurate to say "medaka models"

Please confirm my suspicion and we can change the mentions throughout prior to merging/deploying

thanks for updating to v1.6.2!

erinyoung commented:
I've updated the text in the README to more accurately reflect where the models are from.

kapsakcj commented:
thank you for confirming that and changing the readme! I'll merge and deploy now.

kapsakcj commented:
deploy workflow here: https://github.com/StaPH-B/docker-builds/actions/runs/14073755541

should be available on dockerhub and quay shortly. thanks for the PR!
